### PR TITLE
LPS-103750 & LPS-103753 Show warning errors in sidebar

### PR DESCRIPTION
**Pull request by dgarciasarai**
Tasks to do:
- [x] Remove disabled from "Review and Run" button
- [x] "Review and Run" button validates experiment.goal.target and variants.length before showing the modal to set the condifenceLevel and the splits
- [x] When the user removes the click element target, the error message appears
- [x] When the user removes the variant and the length is <= 1, the error message appears
- [x] Add tests

### CODE REVIEW

**Comment by dgarciasarai** [comment]
Ready to refactor and review 🚀 
TODO TESTS!

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by dgarciasarai** [comment]
ci:test:relevant

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by dgarciasarai** [comment]
ci:test:relevant

File: modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/components/SegmentsExperimentsActions.es.js [hunk 1]
```diff
@@ -30,32 +35,26 @@ import {
...
+		reviewExperimentModal,
+		variants,
+		viewExperimentURL
+	} = useContext(StateContext);
 	const dispatch = useContext(DispatchContext);
 
-	const [reviewModalVisible, setReviewModalVisible] = useState(false);
 	const {observer, onClose} = useModal({
-		onClose: () => setReviewModalVisible(false)
+		onClose: () => dispatch(closeReviewAndRunExperiment(false))
```
Comment (andresfulla) [inline modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/components/SegmentsExperimentsActions.es.js]: No need of that `false`. I think

File: modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/components/SegmentsExperimentsActions.es.js [hunk 2]
```diff
@@ -30,32 +35,26 @@ import {
...
-	const readyToRun = _experimentReady(experiment, variants);
-
 	return (
 		<>
 			{experiment.status.value === STATUS_DRAFT && (
 				<ClayButton
 					className="w-100"
-					disabled={!readyToRun}
-					onClick={() => setReviewModalVisible(true)}
+					onClick={() => dispatch(reviewAndRunExperiment(true))}
```
Comment (andresfulla) [inline modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/components/SegmentsExperimentsActions.es.js]: No need for that `true`.

~Move into a handler. Please~

File: modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/state/actions.es.js [hunk 3]
```diff
@@ -32,6 +32,11 @@ export const closeEditionModal = () => ({
 	type: 'EDIT_EXPERIMENT_FINISH'
 });
 
+export const closeReviewAndRunExperiment = payload => ({
```
Comment (andresfulla) [inline modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/state/actions.es.js]: no need for payload

File: modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/state/actions.es.js [hunk 4]
```diff
@@ -49,6 +54,11 @@ export const openEditionModal = payload => ({
 	type: 'EDIT_EXPERIMENT_START'
 });
 
+export const reviewAndRunExperiment = payload => ({
```
Comment (andresfulla) [inline modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/state/actions.es.js]: +1

File: modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/state/reducer.es.js [hunk 5]
```diff
@@ -162,6 +173,39 @@ function _editExperienceStart(state, experiementModalState = {}) {
 	};
 }
 
+function _reviewAndRunExperiment(state) {
```
Comment (andresfulla) [inline modules/dxp/apps/segments/segments-experiment-web/src/main/resources/META-INF/resources/js/state/reducer.es.js:196]: - [ ] We can take a look together and refactor this

**Comment by andresfulla** [comment]
- [x] Error clean up in deletion and experiment status transitions.

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by dgarciasarai** [comment]
ci:test:relevant

**Comment by dgarciasarai** [comment]
Pull request submitted here https://github.com/brianchandotcom/liferay-portal/pull/80939

**Pull request closed without merging**